linera-client/src/chain_listener.rs

```diff
@@ -202,25 +202,37 @@ impl<C: ClientContext> ChainListener<C> {
 
     /// Runs the chain listener.
     #[instrument(skip(self))]
-    pub async fn run(mut self) -> Result<(), Error> {
+    pub async fn run(mut self) -> Result<impl Future<Output = Result<(), Error>>, Error> {
         let chain_ids = {
             let guard = self.context.lock().await;
-            let mut chain_ids = BTreeSet::from_iter(guard.wallet().chain_ids());
-            chain_ids.insert(guard.wallet().genesis_admin_chain());
-            chain_ids
+            let admin_chain_id = guard.wallet().genesis_admin_chain();
+            guard
+                .make_chain_client(admin_chain_id)
+                .synchronize_from_validators()
+                .await?;
```

Comment:
I'd rather call `self.listen_recursively([admin_chain_id])`, and make sure that, too, first syncs, and then returns a future: That way we don't sync twice, redundantly.

Reply:
Maybe we should think about what functions syncs what – b/c `listen_recursively` does a lot of different things that _may_ lead to duplicating the work of the `synchronize_from_validators`.

Reply:
Also in this category @deuszx notes that this single extra synchronization costs us about 30 round-trips at startup.